components/encryption/src/manager/mod.rs

```diff
@@ -347,6 +404,29 @@ impl Dicts {
         };
         self.rotate_key(key_id, data_key, master_key)
     }
+
+    fn rename_dir(&self, src_name: &str, dst_name: &str) -> Result<()> {
```

Comment:
It seems to me the dir_dict needs to be updated as well. 
For example, when a snapshot is applied, the tablet is renamed from tablet_snapshot/123_5_tmp to tablets/123_5.
All the keys in the dir_dict contains the prefix 123_5_tmp. 
Now after the rename, we're still keeping the 123_5_tmp in the dir_dict. 
But when the caller try to decrypt a SST under 123_5, the path will be 123_5/xxx.sst and it won't find the match.

Reply:
Because "tablet_snapshot/123_5_tmp" is a first level directory under the whitelist, it will be encoded using dir ID. So the data keys are all stored using (dir_id, xx.sst). This way the dir renaming only needs to change the dir_id mapping.